Tests/ParserTests/docatch.flint

```diff
@@ -0,0 +1,22 @@
+// RUN: %flintc %s --dump-ast | %FileCheck %s --prefix CHECK-AST
+
+contract Foo {
+  var arr: [Int] = []
+}
+
+Foo :: (any) {
+  public init() {}
+
+  func foo() -> Int {
+    // CHECK-AST: DoCatchStatement
```

Comment:
Would it also make sense to test that the statements within the do-catch were parsed correctly?

Reply:
Will do.